src/vopr_hub/main.go

```
@@ -0,0 +1,853 @@
+	select {
+	case result := <-vopr_completed:
+		if result != nil {
+			log_error("The VOPR did not complete successfully", message_hash)
+		} else {
+			log_info("The VOPR completed successfully", message_hash)
+		}
+	case max_size := <-limited_buffer.size_reached:
+		if max_size {
+			cmd.Process.Kill()
+		}
+		log_info("The VOPR has completed with a liveness bug", message_hash)
+	}
+
+	// All results are stored in the output.logs byte array
+	output.logs = limited_buffer.buffer.Bytes()
+}
+
+// Writes the debug logs and parsed stack trace to a file on disk.
+func create_issue_file(issue_file_name string, output *vopr_output, message_hash []byte) {
+	full_output := append(output.logs, output.stack_trace...)
+	error := os.WriteFile(issue_file_name, full_output, 0666)
+	if error != nil {
+		error_message := fmt.Sprintf("Failed to write to file: %s", error.Error())
+		log_error(error_message, message_hash)
+		panic(error.Error())
+	} else {
+		log_message := fmt.Sprintf("Created file: %s", issue_file_name)
+		log_info(log_message, message_hash)
+	}
+}
+
+// Submits a GitHub issue that includes the debug logs and parsed stack trace.
+func create_github_issue(message vopr_message, output *vopr_output, issue_file_name string) error {
+	body := create_issue_markdown(message, output)
+	if output.seed_passed {
+		body = "Note this seed passed when it was rerun by the VOPR Hub.<br><br>" + body
+	}
+	// Removes the file path from the name.
+	issue_file_name = strings.Replace(issue_file_name, issue_directory+"/", "", 1)
+	issue_contents := fmt.Sprintf(
+		"{ \"title\": \"%s\", \"body\": \"%s\", \"labels\":[] }",
+		issue_file_name,
+		body,
+	)
+	issue := strings.NewReader(issue_contents)
+	post_request, error := http.NewRequest(
+		"POST",
+		repository_url,
+		issue,
+	)
+	if error != nil {
+		log_error("Failed to create the HTTP request for the GitHub API", message.hash[:])
+		panic(error.Error())
```

Why panic instead of just returning the error? (Likewise on line 571).